### PR TITLE
Update to styled-components v4

### PR DESCRIPTION
**Pull request by korbinancell**
As well as update button component to take advantage of v4 functionality.

### CODE REVIEW

File: components/button/anchor-button.jsx [hunk 1]
```diff
@@ -1,16 +1,13 @@
-import React, { PureComponent } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
-import { BaseButton } from './base-button.jsx';
-import * as Styled from './styled.jsx';
+import { Button } from './component.jsx';
```
Comment (dustinsoftware) [inline components/button/anchor-button.jsx]: Let's separate out the restructuring of `base-button` out from the v4 changes. I'm not sold on this approach but don't want to block the update.

**Comment by dustinsoftware** [comment]
Since this is a breaking change I'd like to wait until the next time we need to push out a breaking change (preferably at least a month from now).

File: components/button/anchor-button.jsx (hunk 1 above)
Comment (korbinancell) [reply components/button/anchor-button.jsx]: Sure thing. Do you want a separate pr for updating the button or just remove the commit and keep it out for now?

Comment (dustinsoftware) [reply components/button/anchor-button.jsx]: Remove it for now, open a separate PR for button after the v4 update gets merged

**Comment by dustinsoftware** [comment]
@awesomebob this should be good to go, please review

File: components/input/component.jsx [hunk 2]
```diff
@@ -1,80 +1,84 @@
...
+					type={type || 'text'}
+					autoFocus={autoFocus}
+					readOnly={readOnly}
+					disabled={disabled}
+					value={value || ''}
+					placeholder={placeholder || ''}
+					onChange={this.handleChange}
+					onClick={onClick}
+					onKeyPress={this.handleKeyPress}
+					ref={forwardedRef}
```
Comment (awesomebob) [inline components/input/component.jsx]: To be clear: this `<Input>` component was formerly *not* forwarding the ref, but we are adding that functionality in this PR, correct?

**Comment by awesomebob** [comment]
Is this a usage of `innerRef` that we'd like to remove as well?
https://github.com/Faithlife/styled-ui/blob/master/components/collapse/component.jsx#L33

File: components/input/component.jsx (hunk 2 above)
Comment (awesomebob) [reply components/input/component.jsx]: I don't see that we previously exposed `innerRef` in this component, might that have been for a different component?

**Pull request merged**